### ISSUE TITLE
Support Mnemonic Code for easy key restoration

### ISSUE THREAD

**Issue by yisiliu** [DimensionDev/Maskbook#15]
We should also support Mnemonic Code for easy key restore, as is done in https://github.com/DimensionDev/PriceDoge/blob/master/extension/src/utils/keyStore.ts

_Originally posted by @yisiliu in https://github.com/DimensionDev/Maskbook/issues/11#issuecomment-483958777_

**Comment by yisiliu** [comment]
Shall we consider this as in our next big release? Although current users need to create a new key pair.

**Comment by Tedko** [comment]
Highly recommend get this done b4 DWeb and other Pr events / summit.
Otherwise on media side we have to spend nearly same manpower for noticing
user

**Comment by neruthes** [comment]
Recently it has been increasingly clear that keypair transfer is more important than keypair backup, and keystore backup is also more important than keypair backup. In essence, keypair backup is a partial backup and is therefore not enough to restore everything. For backup purposes, we still need whole-keystore backup; for transfer purposes, text is not the best medium of transfer, especially when at least one device has optical input capability. Alive backup is better than static backup, and almost everyone has at least 2 devices which may have Maskbook support soon if not now.

In conclusion, I would argue that this kind of backup is not what we really want.

**Comment by Tedko** [comment]
It’s more about standard, since naturally we are more close to the so
called Web3 concept (and Web3 foundation, Eth community foundation). I
don’t want to create another standard on that since the key transfer is
already implemented by other company (such as Brave b4) even though their
solution do involve a server. It is very urgent to support the memoric word
if we want to get their support and using the community’s budget to educate
the users. The DWeb is July 20 and Web3 summit is Aug 17. We’d better
complete this feature by then.

**Comment by Tedko** [comment]
Also since these repo are available for public, let’s discuss somewhere
else and put conclusion here, then see if the community wanna join us or
not.

**Comment by Jack-Works** [comment]
If you want to do this breaking change, you should make a vote to ask current users if they want to do this

**Comment by Tedko** [comment]
@Jack-Works will do.  actually we don't even have a good way to let them vote now. will think later. for now let's finish 1. key tsf mechanism 2. Mnemonic Code

**Comment by neruthes** [comment]
Preview <img width="1440" alt="Screen Shot 2019-07-13 at 06 33 44" src="https://user-images.githubusercontent.com/2139261/61168116-31934f00-a538-11e9-949e-9c7d9fc24b2d.png">

**Comment by neruthes** [comment]
@tedko Would you suggest to prioritize mnemonic words over keystore transfer?

**Comment by Tedko** [comment]
Still, Key transfer 1st. The others feature better be now before Web3 summit.

**Comment by yisiliu** [comment]
@Jack-Works

**Comment by Tedko** [comment]
@Jack-Works  updates?

**Comment by Jack-Works** [comment]
we have discussed this

can be done without breaking but need a lot of time

**Comment by Tedko** [comment]
@Jack-Works great. let me know updates. Happy 10/1

**Comment by Jack-Works** [comment]
supported
